### ISSUE TITLE
Support for managed identity on container_registry

### ISSUE THREAD

**Issue by phatcher** [terraform-providers/terraform-provider-azurerm#9955]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As container_registry supports managed identity, can we have an identity block similar to app_service

### New or Affected Resource(s)

* azurerm_container_registry

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "rg" {
  name     = "resourceGroup1"
  location = "West US"
}

resource "azurerm_container_registry" "acr" {
  name                     = "containerRegistry1"
  resource_group_name      = azurerm_resource_group.rg.name
  location                 = azurerm_resource_group.rg.location
  sku                      = "Premium"
  admin_enabled            = false
  georeplication_locations = ["East US", "West Europe"]

  identity {
    type = "SystemAssigned"
  }
}
```

### References

https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication-managed-identity

* #0000

**Comment by jbinko** [comment]
@phatcher Do you need to provide access to ACR for an identity or do you need to be able to specify identity under which ACR is run? identity block above is for identity for ACR under which ACR is run. The link you provided is referring to access which is not related to identity block above. You'd use identity block above to control access to KV for example. For custom encryption key. E.g. https://docs.microsoft.com/en-us/azure/container-registry/container-registry-customer-managed-keys#create-a-registry-with-customer-managed-key

**Comment by implodingduck** [comment]
As a FYI part of https://github.com/terraform-providers/terraform-provider-azurerm/pull/11661 to add custom encryption the identity block needed to be added.

**Comment by runemy** [comment]
I see that the identity block is supported in azurerm_container_registry, but I do not find any way to export the ACR SystemManagedIdentity, like what is possible in azurerm_app_service:

azurerm_app_service.myapp.identity.0.principal_id

**Comment by tombuildsstuff** [comment]
Fixed via #12378
